src/main/java/dukchess/Duke.java

```diff
@@ -0,0 +1,212 @@
+            }
+            nextCommand = commandMatches.group(1);
+            String commandArgs = getValueOrDefault(commandMatches.group(2), "");
+            commandArgs = commandArgs.trim();
+            switch (nextCommand) {
+            case "list":
+                printAddedTasks();
+                continue;
+            case "bye":
+                sayGoodbye();
+                tasks.flushTaskList();
+                return;
+            case "done":
+                int taskId = Integer.parseInt(commandArgs);
+                String taskStatusAdditionOutcome = setAddedTaskStatus(taskId, true);
+                System.out.println(taskStatusAdditionOutcome);
+                break;
+            case "todo":
+                if (commandArgs.length() == 0) {
+                    System.out.println("Oops, todo description cannot be empty :(");
+                    break;
+                }
+                String todoAdditionOutcome = addTodo(commandArgs);
+                System.out.println(todoAdditionOutcome);
+                break;
+            case "deadline":
+                if (commandArgs.length() == 0) {
+                    System.out.println("Oops, deadline description cannot be empty :(");
+                    break;
+                }
+                String[] deadlineArgs = commandArgs.split(" /by ");
+                if (deadlineArgs.length != 2) {
+                    System.out.println("Oops, deadline due date cannot be empty :(");
+                    break;
+                }
+                String deadlineAdditionOutcome = addDeadline(deadlineArgs[0], deadlineArgs[1]);
+                System.out.println(deadlineAdditionOutcome);
+                break;
+            case "event":
+                if (commandArgs.length() == 0) {
+                    System.out.println("Oops, event description cannot be empty :(");
+                    break;
+                }
+                String[] eventArgs = commandArgs.split(" /at ");
+                if (eventArgs.length != 2) {
+                    System.out.println("Oops, event time cannot be empty :(");
+                    break;
+                }
+                String eventAdditionOutcome = addEvent(eventArgs[0], eventArgs[1]);
+                System.out.println(eventAdditionOutcome);
+                break;
+            case "delete":
+                if (commandArgs.length() == 0) {
+                    System.out.println("You have to specify which task to delete!");
+                    break;
+                }
+                try {
+                    Integer taskIdToDelete = Integer.parseInt(commandArgs);
+                    String deletionOutcome = deleteTask(taskIdToDelete);
+                    System.out.println(deletionOutcome);
+                } catch (NumberFormatException numberFormatException) {
+                    System.out.println("Invalid number passed to task deletion command.");
+                }
+                break;
```

Comment:
Some cases do not have the try catch blocks. You could consider abstracting out details of each case into new methods, including try catch blocks for each.